users/users.go

```diff
@@ -57,23 +62,34 @@ type UserRepository interface {
 }
 
 func isEmail(email string) bool {
-	if len(email) < 6 || len(email) > 254 {
+	if email == "" {
+		return false
+	}
+
+	es := strings.Split(email, "@")
+	if len(es) != 2 {
 		return false
 	}
+	local, host := es[0], es[1]
 
-	at := strings.LastIndex(email, "@")
-	if at <= 0 || at > len(email)-3 {
+	if len(local) == 0 || len(host) == 0 {
```

Comment:
We are not checking the extension here. Or you revert this `len(host) - 3 == 0` or you can maybe do an extra Split to know the real size of the extension.